main.go

```diff
@@ -138,11 +156,53 @@ func walk(ch chan<- *file, start string) {
 }
 
 func addLicense(path string, fmode os.FileMode, tmpl *template.Template, data *copyrightData) (bool, error) {
+	var lic []byte
+	var err error
+	lic, err = licenseHeader(path, tmpl, data)
+	if err != nil || lic == nil {
+		return false, err
+	}
+
+	b, err := ioutil.ReadFile(path)
+	if err != nil || hasLicense(b) {
+		return false, err
+	}
+
+	line := hashBang(b)
+	if len(line) > 0 {
+		b = b[len(line):]
+		if line[len(line)-1] != '\n' {
+			line = append(line, '\n')
+		}
+		lic = append(line, lic...)
+	}
+	b = append(lic, b...)
+	return true, ioutil.WriteFile(path, b, fmode)
+}
+
+func fileHasLicense(path string, tmpl *template.Template, data *copyrightData) (bool, error) {
```

Comment:
Feels like `fileHasLicense` checks whether the given file at `path` contains a specific license defined by `tmpl` and `data` but I don't believe it does. It only needs those args because it calls `licenseHeader`.

Ideally, I'd love to see this signature:

```go
// fileHasLicense reports whether the file at path contains a license header.
func fileHasLicense(path string) (bool, error)
```

Unless you really want to check whether a file contains exactly that specific license but it would be a different feature and might produce a regression.